lib/embed.dart

```diff
@@ -790,7 +803,57 @@ class Embed extends EditorUi {
   void displayIssues(List<AnalysisIssue> issues) {
     testResultBox.hide();
     hintBox.hide();
-    analysisResultsController.display(issues);
+
+    // Handle possiblity of issues in appended test code.
+    analysisResultsController
+        .display(detectIssuesInTestSourceAndModifyIssuesAccordingly(issues));
+  }
+
+  // We append test source code to the user's source code, because of
+  // this we possibly have a special situation..
+  // There could be warnings or errors in the *TEST* code that is being
+  // appended to the user's dart source.
+  // This can result in issues with line numbers that are
+  // outside the user's dart source.  This would confusing to the users.
+  // We are going to do one of two things:
+  // - If the test source is currently HIDDEN and the issue kind is
+  // not and `error` (it is `info` or `warning`) then we will REMOVE
+  // the issue from the list so as to "hide" it.
+  // - If the test source is showing, *or* if the issue is an `error`, we are
```

Comment:
> *or* if the issue is an `error`

What about info level and warning-level issues? Shouldn't we adjust the line number for those too?

Reply:
I am not sure if I follow.. ?   (If the info/warning are being shown they do get their line numbers adjusted, and if not they are removed (line numbers set to -99 are removed below..))